Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -623,6 +624,9 @@ gd::String EventsCodeGenerator::GenerateParameterCodes(
   // Code only parameter type
   else if (metadata.type == "inlineCode") {
     argOutput += metadata.supplementaryInformation;
+  } else if(metadata.type == "uid") {
+    // GenerateSingleUsageUniqueIdFor would be better, but wouldn't work with expressions.
+    argOutput += ConvertToStringExplicit(gd::UUID::MakeUuid4());

```

Comment:
Not a big fan of this as there is a chance of collisions even if little, and it will break hot reload. I cannot use the better `GenerateSingleUsageUniqueIdFor` though, as it only accepts an instruction. As far as I saw, it doesn't actually care about  the type of the parameter only about the pointer value, should I change it to accept a void pointer so that I can use an expression to identify it too?

Reply:
What about another overload of this function?

Reply:
That means we would have to overload the GenerateParameterCode and GenerateParametersCode with the exact same implementation just different types :/

Reply:
Wouldn't that be too much bloat and confusing?

Reply:
Depends, there is a few things to consider:
- is the pointer to the expression stable across exports?
- if we move in the events editor a condition to another event, is this pointer still stable?

If yes, we could use this pointer. Otherwise, we could maybe use a mix of the pointer of the instruction + something like the parameter index.

Reply:
(also, the reason for making something not using void* is to explicitly generate these questions and conversation - do we really need to do that? Is there something else to uniquely identify something that is not using the memory pointer?)

Reply:
> is the pointer to the expression stable across exports?

Oh right, the expression pointer probably isn't stable. Not quite sure what I should do here.

Reply:
I digged a bit more and couldn't find any object (or at least easy to access ones) that would work with expressions. Should I just disallow that parameter on expressions?

Reply:
Maybe it's just simpler to only allow this on instructions yeah!